All zero check

LevyNat opened the pull request:
In the Hotspot init, there is a check to ensure that a gene is not 'all zero'. i..e. never expressed in any cell. 

It checks that the sum of its expression over all cells is greater than 0, which makes sense if we consider NB or Bernoulli data; but if we have 'normal' or 'none' (meaning no assumption about data distribution but assuming data is already standardized) one can get negative or zero sums.  For instance this is the case if one wants to compute the autocorrelation of module scores, which are PC loadings and real numbers. 

Thus I propose to change the valid_genes computation to consider an 'all zero' column iff all elements are zero.

deto commented:
Thanks - it makes sense that this check isn't expansive enough.  However, the real issue is that the std. deviation for genes that are all 0 is zero and then this leads to nans/infinities later on.  Could you change the PR so that instead of checking for all zero, you check to see if the std of the gene == 0?  I think that should cover all cases better.

LevyNat commented:
Thanks @deto, I updated to check if a gene row is constant, which means it has 0 std. 

In addition there is a warning for the use of a sparse matrix different from csr format but the warning text said we should use a csc matrix. I updated to csr, is it correct?

deto reviewed (approved):
Looks good to me
